_posts/2015-12-03-journey_to_high_availability.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+title: Journey to High Availability
+date: 2015-12-3 11:43:55
+author: Christopher Pisano - The Advisory Board Company
+tags:
+- foreman
+- puppet
+- high_availability
+guest_post: true
+excerpt: |
+  This is a recount of the road from an all in one Foreman implementation to a full HA deployment. 
+  We will also be looking forward beyond our current implementation to future plans.
+---
+
+###Overview
+
+We started with Foreman almost two years ago now because we simply needed a front end to Puppet and the Open Source Puppet Dashboard just didn't fit the bill at the time. We deployed an all in one implementation as a POC using MySQL and eventually moved into production with a PostgreSQL database and two clustered external Puppet Masters. The Puppet CA lived on Foreman as well as the Smart-Proxy and we set up an unnecessary host group structure with an obscene amount of Smart-Class Parameters; we did not use Hiera at the time. As our environment grew our implementation became slower and slower. We added resources but at the end of the day it was still a single instance running the Puppet CA, Foreman webUI and the ENC/Reports functions. As we grew to over 1,000 hosts (today ~1,800) things got worse; our host groups increased, our Smart-Class Parameters increased and our performance decreased. We lived with this for some time being satisfied with the status quo until we set an initiative to deploy an implementation suitable for an enterprise. It is here where all the fun begins.
+
+[Foreman HA Case Study](https://www.youtube.com/watch?v=bJ5a67SSA-s)
```

Comment:
Maybe just preface this with a line of explanation? "If you want to see a live discussion of this topic, here's an interview as part of the Foreman Case Study series" or similar?